tee-worker/litentry/core/assertion-build/src/generic_discord_role.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020-2023 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+use crate::*;
+use lc_credentials::{generic_discord_role::GenericDiscordRoleAssertionUpdate, Credential};
+use lc_data_providers::{
+	discord_litentry::DiscordLitentryClient, DataProviderConfigReader, ReadDataProviderConfig,
+};
+use lc_stf_task_sender::AssertionBuildRequest;
+use litentry_primitives::GenericDiscordRoleType;
+
+pub fn build(req: &AssertionBuildRequest, rtype: GenericDiscordRoleType) -> Result<Credential> {
+	let role_id = get_generic_discord_role_id(&rtype).map_err(|error_detail| {
+		Error::RequestVCFailed(Assertion::GenericDiscordRole(rtype.clone()), error_detail)
+	})?;
+
+	let mut has_role_value = false;
+	let mut client = DiscordLitentryClient::new();
+	for identity in &req.identities {
+		if let Identity::Discord(address) = &identity.0 {
+			let resp =
+				client.has_role(role_id.clone(), address.inner_ref().to_vec()).map_err(|e| {
+					Error::RequestVCFailed(
+						Assertion::GenericDiscordRole(rtype.clone()),
+						e.into_error_detail(),
+					)
+				})?;
+
+			debug!("Litentry & Discord user has role response: {:?}", resp);
+
+			if resp.data {
```

Comment:
I see the `DiscordResponse` has multiple fields:
```
pub struct DiscordResponse {
	pub data: bool,
	pub message: String,
	pub has_errors: bool,
	pub msg_code: u32,
	pub success: bool,
}
```
`data` is what we look for, right? I mean other fields like `has_errors` and `success` could be confusing
cc @zhouhuitian too

Reply:
I mean do we have to check more fields? Like `success`? (to make sure it's a valid response - that's my guess for this field tho)

Reply:
1 Success field has been removed long time ago from tee-microservice, so this will be removed too here.

2 There is no need to check other fields like 'hasErrors' because data is always false  if 'hasErrors' is true.
And the data is true if the user has the specified role,

See the true case: https://github.com/litentry/tee-microservice/blob/7e42df67e55af17f2a9d07281bb3158d65026ec3/src/main/java/com/litentry/litbot/TEEBot/restservice/VerifyMsgResource.java#L174

And false case:
https://github.com/litentry/tee-microservice/blob/7e42df67e55af17f2a9d07281bb3158d65026ec3/src/main/java/com/litentry/litbot/TEEBot/restservice/VerifyMsgResource.java#L160-L163

And default hasError as false:
https://github.com/litentry/tee-microservice/blob/7e42df67e55af17f2a9d07281bb3158d65026ec3/src/main/java/com/litentry/litbot/TEEBot/restservice/InvokeResult.java#L44